### PR TITLE
Add tenant id to lambda context and structured log messages

### PR DESCRIPTION
**Pull request by ramisa2108**
*Issue #, if available:*

*Description of changes:*
- Add tenant id to LambdaContext object and structured log messages 

*Target (OCI, Managed Runtime, both):*
- Both

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/runtimeapi/LambdaRuntimeApiClientImplTest.java [hunk 1]
```diff
@@ -333,6 +334,31 @@ public void nextTest() {
         }
     }
 
+    @Test
```
Comment (maxday) [inline aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/runtimeapi/LambdaRuntimeApiClientImplTest.java:337]: great test! 💯

File: aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/EventHandlerLoader.java [hunk 2]
```diff
@@ -581,7 +581,8 @@ public ByteArrayOutputStream call(InvocationRequest request) throws Error, Excep
                         cognitoIdentity,
                         LambdaEnvironment.FUNCTION_VERSION,
                         request.getInvokedFunctionArn(),
-                        clientContext
+                        clientContext,
```
Comment (maxday) [inline aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/EventHandlerLoader.java:584]: I think it would be great to leave the clientContext as the last parameter? This way we keep aws-related fields grouped. WDYT?

File: aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/api/LambdaContext.java [hunk 3]
```diff
@@ -34,7 +35,8 @@ public LambdaContext(
             CognitoIdentity identity,
             String functionVersion,
             String invokedFunctionArn,
-            ClientContext clientContext
+            ClientContext clientContext,
```
Comment (maxday) [inline aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/api/LambdaContext.java:38]: same here, maybe it would be better to swap args

File: aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/runtimeapi/LambdaRuntimeApiClientImplTest.java [hunk 4]
```diff
@@ -333,6 +334,31 @@ public void nextTest() {
...
+    public void nextWithTenantIdTest() {
+        try {
+            MockResponse mockResponse = new MockResponse();
+            mockResponse.setResponseCode(HTTP_ACCEPTED);
+            mockResponse.setHeader("lambda-runtime-aws-request-id", "1234567890");
+            mockResponse.setHeader("Content-Type", "application/json");
+            String expectedTenantId = "my-tenant-id";
+            mockResponse.setHeader("lambda-runtime-aws-tenant-id", expectedTenantId);
```
Comment (maxday) [inline aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/runtimeapi/LambdaRuntimeApiClientImplTest.java:345]: I think it would be great to have another couple of tests for:
1. header not present
2. header present and empty string
3. header present and null value

File: aws-lambda-java-runtime-interface-client/src/main/jni/deps/aws-lambda-cpp-0.2.7/include/aws/lambda-runtime/runtime.h [hunk 5]
```diff
@@ -61,6 +61,11 @@ struct invocation_request {
      */
     std::chrono::time_point<std::chrono::system_clock> deadline;
 
+    /**
```
Comment (maxday) [inline aws-lambda-java-runtime-interface-client/src/main/jni/deps/aws-lambda-cpp-0.2.7/include/aws/lambda-runtime/runtime.h:64]: hopefully at some point, we won't have a copy of aws-lambda-cpp in here, so let's keep that change here but I think you also need to modify the source of truth here: https://github.com/awslabs/aws-lambda-cpp?

**Review by maxday — COMMENTED** [review]
Awesome job @ramisa2108 ! 💯 
I left only some small comments, let me know if that makes sense

File: aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/logging/JsonLogFormatterTest.java [hunk 6]
```diff
@@ -29,7 +29,8 @@ void testFormattingWithLambdaContext() {
                 null,
                 null,
                 "function-arn",
-                null
+                null,
+                "tenant-id"
```
Comment (maxday) [inline aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/logging/JsonLogFormatterTest.java:33]: let's also add a test with a null value for tenant-id as it's optional

File: aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/EventHandlerLoader.java (hunk 2 above)
Comment (ramisa2108) [reply aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/EventHandlerLoader.java]: Great suggestion. Updated.

File: aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/api/LambdaContext.java (hunk 3 above)
Comment (ramisa2108) [reply aws-lambda-java-runtime-interface-client/src/main/java/com/amazonaws/services/lambda/runtime/api/client/api/LambdaContext.java]: Updated.

File: aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/runtimeapi/LambdaRuntimeApiClientImplTest.java (hunk 4 above)
Comment (ramisa2108) [reply aws-lambda-java-runtime-interface-client/src/test/java/com/amazonaws/services/lambda/runtime/api/client/runtimeapi/LambdaRuntimeApiClientImplTest.java:334]: Thanks for the suggestion. Added these test cases.

**Review by maxday — APPROVED** [review]
🎉

File: aws-lambda-java-runtime-interface-client/src/main/jni/deps/aws-lambda-cpp-0.2.7/include/aws/lambda-runtime/runtime.h (hunk 5 above)
Comment (godcrampy) [reply aws-lambda-java-runtime-interface-client/src/main/jni/deps/aws-lambda-cpp-0.2.7/include/aws/lambda-runtime/runtime.h:64]: Agreed, one day, we could get rid of this. (Not blocking PR from my end either).

**Review by godcrampy — APPROVED** [review]
Great work! 🎉
